main.nf

```diff
@@ -122,14 +130,17 @@ if (!params.fastq) {
         output:
             file 'fastq.fq.gz' into fastq
 
-        publishDir 'results'
-
         module 'bioinfo-tools'
         module "$params.modules.samtools"
 
         // We only need one core for this part
-        clusterOptions = {
-            "-A $params.project -p $params.runspecs.core $params.runspecs.extra"
+        if ( nextflow_running_as_slurmjob() ) {
```

Comment:
Process directives have to be called in the process block, I can't, unfortunately put that in a subroutine. So we have to live with some code duplication (unless I write a macro for it, then it should work I guess, but I don't know how to write groovy macros).